NCLtomakeSCRIP.ncl

```diff
@@ -0,0 +1,154 @@
+ 
+
+;static missing data points?
+MissingData = False           
+MaskName_user = "NA" ;if True, what would be an appropriate name for the mask?
+;MaskName = "landmask"
+
+;note that for missing values changing with time, we need a different approach to regrid
+; in that case I don't create static masks.
+; https://nco.sourceforge.net/nco.html#index-_005fFillValue
+; https://www.ncl.ucar.edu/Applications/ESMF.shtml#WeightsAndMasking
+
+if(MissingData) then
+    MaskName = MaskName_user
+    tgtSCRIPfile = ("SCRIP_" + gridname + "_" + MaskName + ".nc")
+else
+    MaskName = default_fillvalue("string")
+end if
+
+
+;----------------------------------------------------------------------
+;  Generate a description file (SCRIP) for the target grid.
+;----------------------------------------------------------------------
+
+print("gridname : " + gridname)
+
+ESMFOpt                = True
+ESMFOpt@ForceOverwrite = True
+ESMFOpt@Overwrite       = True
+ESMFOpt@PrintTimings   = True
+ESMFOpt@Debug          = True
+
+sfile = addfile((inputdir + "/" + latlonfile),"r") 
+;read the input netcdf file and create a file object sfile
+
+if(MissingData) then ; Necessary if x has STATIC missing values.
+    ;change the output file name
+    print("taking care of static missing masks")
+
+    tgtSCRIPfile = ("SCRIP_" + tgtres + "_" + MaskName + ".nc")
+    
+    ;an example for for ERA5 land-sea mask
+    if(gridname .eq. "ERA5_0.25deg") then
+        ;read an example variable with missing values 
+        svar  = rm_single_dims(sfile->LSM(0,:,:))  ; do not include time or vertical dimensions
+        
+        if(MaskName .eq. "ocnmask") then
+            itemp = svar ; oceanmask
+        else
+            itemp = where(svar .lt. 1.0, 1 , 0) 
+            copy_VarCoords(svar,itemp)  ;landmask
+        end if
+        
+    end if
+    
+    
+    ESMFOpt@GridMask  = itemp     
+    
+end if
+;warning:NetOpenFile: MissingToFillValue option set True, but missing_value attribute and data variable (evapr) types differ: not adding virtual _FillValue attribute
+; read in the latitude and longitude arrays 
+ilat = sfile->$latname$
+ilon = sfile->$lonname$
+
+sdims = dimsizes(ilat)
+ndims = dimsizes(sdims)
+
+; need to remove any unecessary dimensions
+; such as time in the WRF output
+; assuming such additional dimensions has a size of 1
+
+if(GridType .eq. "curvilinear") 
+    if (ndims .gt. 2) then
+        ilat := rm_single_dims(ilat) ; the ":=" operate overwrites the variable
+        ilon := rm_single_dims(ilon) 
+    end if
+else
+    if(ndims .gt. 1) then
+        ilat := rm_single_dims(ilat) 
+        ilon := rm_single_dims(ilon) 
+    end if
+    
+end if
+
+print("running ESMF function")
+; run the ESMF functions, defined for different types of grid
+if(GridType .eq. "latlon") then   ; uniform lat&lon grid, need to provide grid type
+     print("calling latlon_to_SCRIP")
+     latlon_to_SCRIP((tgtSCRIPdir + "/" + tgtSCRIPfile),latlongridname,ESMFOpt) 
```

Comment:
I'm getting an error that the variable `latlongridname` is undefined when I try to use the `lat-lon` grid type. Is that an indication that I am missing some sort of input?